cmd/preview/list.go

```diff
@@ -72,16 +88,66 @@ func executeListPreviews(ctx context.Context, opts ListFlags) error {
 		}
 		return fmt.Errorf("failed to get preview environments: %s", err)
 	}
-	w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
-	fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+	switch opts.output {
+	case "json":
+		previewListOutput, err := getPreviewOutput(ctx, opts, oktetoClient)
+		if err != nil {
+			return err
+		}
+		bytes, err := json.MarshalIndent(previewListOutput, "", " ")
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	case "yaml":
+		previewListOutput, err := getPreviewOutput(ctx, opts, oktetoClient)
+		if err != nil {
+			return err
+		}
+		bytes, err := yaml.Marshal(previewListOutput)
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	default:
+		w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
+		fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+		for _, preview := range previewList {
+			previewLabels := "-"
+			fmt.Fprintf(w, "%s\t%s\t%v\t%s\n", preview.ID, preview.Scope, preview.Sleeping, previewLabels)
+		}
+		w.Flush()
+	}
+	return nil
+}
+
+func getPreviewOutput(ctx context.Context, opts ListFlags, oktetoClient types.OktetoInterface) ([]PreviewOutput, error) {
+	var previewSlice []PreviewOutput
+	previewList, err := oktetoClient.Previews().List(ctx, opts.labels)
+	if err != nil {
+		if uErr, ok := err.(oktetoErrors.UserError); ok {
+			return nil, uErr
+		}
+		return nil, fmt.Errorf("failed to get preview environments: %s", err)
+	}
 	for _, preview := range previewList {
 		previewLabels := "-"
-		if len(preview.PreviewLabels) > 0 {
-			previewLabels = strings.Join(preview.PreviewLabels, ", ")
+		previewOutput := PreviewOutput{
+			Name:     preview.ID,
+			Scope:    preview.Scope,
+			Sleeping: preview.Sleeping,
+			Labels:   previewLabels,
```

Comment:
```suggestion
			Labels:   preview.PreviewLabels.,
```
we are always set `-` as value